pre-registration/pre-registration-application-service/src/main/java/io/mosip/preregistration/application/config/CacheConfig.java

```diff
@@ -1,41 +1,61 @@
 package io.mosip.preregistration.application.config;
 
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.cache.CacheManager;
 import org.springframework.cache.annotation.CachingConfigurerSupport;
 import org.springframework.cache.annotation.EnableCaching;
 import org.springframework.cache.ehcache.EhCacheCacheManager;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.core.env.Environment;
 
 import net.sf.ehcache.config.CacheConfiguration;
 
 @EnableCaching
 @Configuration
 public class CacheConfig extends CachingConfigurerSupport {
 
-	@Autowired
-	private Environment env;
+	@Value("${preregistration.cache.memoryEvictionPolicy}")
+	private String memoryEvictionPolicy;
+
+	@Value("${preregistration.cache.maxEntriesLocalHeap}")
+	private long maxEntriesLocalHeap;
+
+	@Value("${preregistration.cache.setMaxBytesLocalDisk}")
+	private String setMaxBytesLocalDisk;
+
+	@Value("${preregistration.cache.TimeToLiveSeconds}")
+	private long TimeToLiveSeconds;
+
+	@Value("${preregistration.cache.setlogging}")
+	private boolean setLogging;
+
+	@Value("${preregistration.cache.isEternal}")
+	private boolean isEternal;
 
 	@Bean
 	public net.sf.ehcache.CacheManager ehCacheManager() {
```

Comment:
why is the ehcache hard coded here? Can you speak to mono and see how he uses it?

Reply:
it used to differentiate between the cacheManger of springframework and ehcache , In packmanager there is implementation for  cache using the default one from springframework